mllib/src/main/scala/org/apache/spark/ml/image/HadoopUtils.scala

```diff
@@ -103,16 +102,16 @@ private object SamplePathFilter {
       // scalastyle:off hadoopconfiguration
       val hadoopConf = spark.sparkContext.hadoopConfiguration
       // scalastyle:on hadoopconfiguration
-      val old = Option(hadoopConf.getClass(flagName, null))
+      val old = hadoopConf.getClass(flagName, null)
```

Comment:
Not a big deal but I though option and pattern match with Some and None is fine.

Reply:
Same sort of issue: assigning `Option[Class[_]]` caused the warning. Here it seemed as fine to stick with the underlying null vs not-null.